runtime/autoload/man.vim

```diff
@@ -118,15 +118,12 @@ endfunction
 " Expects a string like 'access' or 'access(2)'.
 function s:parse_page_and_section(sect, str) abort
   try
-    let save_isk = &iskeyword
-    setlocal iskeyword-=(,)
-    let page = substitute(a:str, '(*\(\k\+\).*', '\1', '')
-    let sect = substitute(a:str, '\(\k\+\)(\([^()]*\)).*', '\2', '')
+    let page = substitute(a:str, '\([-.[:alnum:]_]\+\).*', '\1', '')
+    let sect = substitute(a:str, '\([-.[:alnum:]_]\+\)(\([^()]*\)).*', '\2', '')
     if sect == page || -1 == match(sect, '^[0-9 ]\+$')
```

Comment:
If rendered as

      let [page, sect] = matchlist(a:str, '\v\C([-.[:alnum:]_]+)%(\((\d+)\))?')[1:2]

the second part of the condition can be removed—`sect` is guaranteed to either be empty or contain only numeric characters.

This works for various inputs:

- `:Man man`
- `:Man man(1)`
- `:Man git-branch`
- `:Man foo.bar`

If that looks good, I will update the commit.